application/marathon.go

```diff
@@ -71,50 +71,60 @@ func (m *MarathonFinder) Apps() (Apps, error) {
 	apps := map[string]App{}
 
 	for _, a := range ma {
-		name := a.Labels["zoidberg_app_name"]
-		if name == "" {
-			log.Printf("app %s has no label zoidberg_app_name\n", a.ID)
-			continue
-		}
+		meta := parseLabels(a.Labels)
+		for port, labels := range meta {
+			if labels["balanced_by"] != m.balancer {
+				continue
+			}
+			name := labels["app_name"]
+			if name == "" {
+				log.Printf("app %s has no label zoidberg_app_name\n", a.ID)
+				continue
+			}
 
-		version := a.Labels["zoidberg_app_version"]
-		if version == "" {
-			version = "1"
-		}
+			version := labels["app_version"]
+			if version == "" {
+				version = "1"
+			}
 
-		app := apps[name]
-		if app.Name == "" {
-			app.Name = name
-			app.Servers = []Server{}
-			app.Meta = metaFromLabels(a.Labels)
-		}
+			app := apps[name]
+			if app.Name == "" {
+				app.Name = name
+				app.Servers = []Server{}
 
-		for _, task := range a.Tasks {
-			healthy := true
-			for _, check := range task.HealthCheckResult {
-				if check == nil {
-					continue
+				// labels only come from the first task,
+				// this could lead to funny errors if there
+				// are multiple tasks with the same zoidberg_app_name
```

Comment:
This comment only applies to Mesos finder, Marathon provides labels on app level